Mathlib/Logic/Equiv/LocalEquiv.lean

```diff
@@ -0,0 +1,36 @@
+import Mathlib.Init.Logic
+import Mathlib.Init.Set
+import Mathlib.Logic.Equiv.MfldSimpsAttr
+import Mathlib.Tactic.Core
+
+open Lean Meta Elab Tactic
+
+/-! Implementation of the `mfld_set_tac` tactic for working with the domains of partially-defined
+functions (`LocalEquiv`, `LocalHomeomorph`, etc).
+
+This is in a separate file from `Mathlib.Logic.Equiv.MfldSimpsAttr` because attributes need a new
+file to become functional.
+-/
+
+namespace Tactic.MfldSetTac
+
+theorem Set.ext {α : Type u} {a b : Set α} (h : ∀ (x : α), x ∈ a ↔ x ∈ b) : a = b :=
+funext (fun x => propext (h x))
```

Comment:
Can this just go in Mathlib.Init.Set?

Reply:
Sure.  Do we need to add a comment there mentioning that this lemma was added because it was used for a certain tactic?